### PR TITLE
feat: adds question module

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Introduces router scaffolding for inbound Question module. The intention here is to establish the foundations and then divide the work across multiple tickets for parallelisation. For example:  Create, Edit, Commenting and display work.

This is being introduced as a standalone module. It will be enabled by default for any installs which **do not declare** the modules they use. As all production instances specify their modules with the `REACT_APP_SUPPORTED_MODULES` configuration option this means the new Questions module will not be "leaked" to public sites.

### Outstanding

- [x] Tests – it should be easy for people to introduce tests for Question functionality, running using testing-library to speed up the feedback loop
- [x] Store and collections

Placeholder Routes

Screenshot 2023-09-15 at 09 34 47
Screenshot 2023-09-15 at 09 34 57

#2859

### CODE REVIEW

**Review by felipelmota — APPROVED** [review]

File: src/pages/Question/question.routes.test.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  setActiveResearchItemBySlug = jest.fn()
+  needsModeration = jest.fn().mockResolvedValue(true)
+  incrementViewCount = jest.fn()
+  activeResearchItem = FactoryQuestionItem({
+    title: 'Question title',
+    _createdBy: 'jasper',
+  })
+  researchUploadStatus = {} as any
+  updateUploadStatus = {} as any
+  formatResearchCommentList = jest.fn()
```
Comment (AlfonsoGhislieri) [inline src/pages/Question/question.routes.test.tsx:46]: **Issue**: many of these mocked functions seem to be for the Research store and not the question store.

File: src/stores/Question/question.store.tsx [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+  Complete: false,
+})
+
+/**
+ * Export an empty context object to be shared with components
+ * The context will be populated with the QuestionStore in the module index
+ * (avoids cyclic deps and ensure shared module ready)
+ */
+export const QuestionStoreContext = createContext<QuestionStore>(null as any)
+export const useQuestionStore = () => useContext(QuestionStoreContext)
```
Comment (AlfonsoGhislieri) [inline src/stores/Question/question.store.tsx:317]: **Suggestion**: would be good to add tests for the question store

**Review by AlfonsoGhislieri — CHANGES_REQUESTED** [review]

File: src/models/question.models.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import type { DBDoc, IModerable, ISharedFeatures } from '.'
+
+/**
+ * Research retrieved from the database also include metadata such as _id, _created and _modified
+ */
+export type IQuestionDB = DBDoc & IQuestion.Item
+
+/** All typings related to the Research Module can be found here */
+type UserIdList = string[]
+
```
[the remark this answers is not in the document]
Comment (thisislawatts) [reply src/models/question.models.tsx:10]: Perhaps but that seems like a refactor that would benefit a broader chunk of the codebase and is not tightly coupled to the change we are making here.

File: src/models/question.models.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  } & Omit<FormInput, 'collaborators'> &
+    ISharedFeatures
+
+  export interface FormInput extends IModerable {
+    title: string
+    description: string
+    slug: string
+    previousSlugs?: string[]
+  }
+}
```
[the remark this answers is not in the document]
Comment (thisislawatts) [reply src/models/question.models.tsx:26]: Resolved in e8a6c4faa5739b507aa7a8916e4cb085cfb078fb

File: src/stores/Question/question.store.tsx (hunk 2 above)
Comment (thisislawatts) [reply src/stores/Question/question.store.tsx:88]: Fixed in: 34899bae33b263c722981da56d9c17f4808df28f

**Review by AlfonsoGhislieri — APPROVED** [review]
Thanks for the changes and adding the tests and the scaffolding for store testing @thisislawatts

**Pull request merged**